File: DuDoong-Domain/src/main/java/band/gosrock/domain/common/vo/IssuedTicketOptionAnswerVo.java
package band.gosrock.domain.common.vo;


import band.gosrock.domain.domains.issuedTicket.domain.IssuedTicketOptionAnswer;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class IssuedTicketOptionAnswerVo {

    private final Long issuedTicketOptionAnswerId;

    private final String optionQuestion;

    private final String answer;

    private final Money additionalPrice;

    public static IssuedTicketOptionAnswerVo from(
            IssuedTicketOptionAnswer issuedTicketOptionAnswer) {
        return IssuedTicketOptionAnswerVo.builder()
                .issuedTicketOptionAnswerId(issuedTicketOptionAnswer.getId())
                .answer(issuedTicketOptionAnswer.getAnswer())
                .additionalPrice(issuedTicketOptionAnswer.getAdditionalPrice())
                .build();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/common/vo/IssuedTicketOptionAnswerVo.kt
package band.gosrock.domain.common.vo

import band.gosrock.domain.domains.issuedTicket.domain.IssuedTicketOptionAnswer

data class IssuedTicketOptionAnswerVo(
    val issuedTicketOptionAnswerId: Long? = null,
    val optionQuestion: String? = null,
    val answer: String? = null,
    val additionalPrice: Money? = null,
) {
    companion object {
        @JvmStatic
        fun from(issuedTicketOptionAnswer: IssuedTicketOptionAnswer): IssuedTicketOptionAnswerVo =
            IssuedTicketOptionAnswerVo(
                issuedTicketOptionAnswerId = issuedTicketOptionAnswer.id,
                answer = issuedTicketOptionAnswer.answer,
                additionalPrice = issuedTicketOptionAnswer.additionalPrice,
            )
    }
}
